### PR TITLE
Add ecryptfs check for overlay2

### PR DESCRIPTION
**Pull request by HackToday**
We added docs https://github.com/docker/docker/commit/a546042b91f655c7cf53484cdb0c5c8b3cf12d33
about ecryptfs check but not in code side.
Also refactor code to make it clean.

Signed-off-by: Kai Qiang Wu(Kennan) wkqwu@cn.ibm.com

### CODE REVIEW

**Comment by vdemeester** [comment]
@HackToday `vet` error

```
daemon/graphdriver/overlay2/overlay.go:120: possible formatting directive in Error call
```

**Comment by HackToday** [comment]
Win2lin  always failed with

```
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: fork/exec /usr/bin/gcc: cannot allocate memory
```

Not know why :0

**Comment by HackToday** [comment]
please test it

**Comment by HackToday** [comment]
retest this please

File: daemon/graphdriver/overlay2/overlay.go [hunk 1]
```diff
@@ -116,17 +116,8 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 
 	// check if they are running over btrfs, aufs, zfs or overlay
```
Comment (dmcgowan) [inline daemon/graphdriver/overlay2/overlay.go]: this comment should be updated as well

**Comment by dmcgowan** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

File: daemon/graphdriver/overlay2/overlay.go [hunk 2]
```diff
@@ -116,17 +116,8 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
...
-	case graphdriver.FsMagicAufs:
-		logrus.Error("'overlay' is not supported over aufs.")
-		return nil, graphdriver.ErrIncompatibleFS
-	case graphdriver.FsMagicZfs:
-		logrus.Error("'overlay' is not supported over zfs.")
-		return nil, graphdriver.ErrIncompatibleFS
-	case graphdriver.FsMagicOverlay:
-		logrus.Error("'overlay' is not supported over overlay.")
+	case graphdriver.FsMagicBtrfs, graphdriver.FsMagicAufs, graphdriver.FsMagicZfs, graphdriver.FsMagicOverlay, graphdriver.FsMagicEcryptfs:
+		logrus.Errorf("'overlay' is not supported over %s", backingFs)
```
Comment (thaJeztah) [inline daemon/graphdriver/overlay2/overlay.go]: should this error use "overlay2" here?

Comment (dmcgowan) [inline daemon/graphdriver/overlay2/overlay.go]: I think you are probably right, `'overlay'` in single quotes is intended to refer to the graph driver name and not the filesystem name.

**Comment by tiborvass** [comment]
Sorry, if someone wants to send another pr to fix it, feel free

**Comment by thaJeztah** [comment]
@tiborvass np, we just crossed each other, I'll open a PR

**Comment by HackToday** [comment]
I  did notice the overlay2 for message, but I thought may be @dmcgowan wants to make other think overlay2 still overlay technology, so I did not change that name. Which overlay maybe a general 😆

**Comment by thaJeztah** [comment]
@HackToday no worries; I thought it would be more consistent to show from which driver the error came, but I understand the thought

**Pull request merged**